### PR TITLE
[SM-361] Add Support for never expiring ApiKeys

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This PR is to add support for never expiring access-tokens for the secrets manager project.

## Code changes

- src/Api/Models/Response/SecretsManager/AccessTokenResponseModel.cs
src/Api/SecretManagerFeatures/Models/Request/AccessTokenCreateRequestModel.cs
src/Api/SecretManagerFeatures/Models/Response/AccessTokenCreationResponseModel.cs

Update request and response models for the Api to support null `ExpireAt`.

- test/Api.Test/Controllers/ServiceAccountsControllerTests.cs

Fix naming for unit test

- test/Api.IntegrationTest/Controllers/ServiceAccountsControllerTests.cs

Add integration test for using null expireDate.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by Thomas-Avery** [comment]
After this is reviewed, I will generate EF migrations in this PR.

**Review by coltonhurst — COMMENTED** [review]
This looks great! Only one small thing.

File: src/Api/SecretManagerFeatures/Models/Request/AccessTokenCreateRequestModel.cs [hunk 1]
```diff
@@ -17,8 +17,7 @@ public class AccessTokenCreateRequestModel
     [Required]
     [EncryptedString]
     public string Key { get; set; }
-    [Required]
-    public DateTime ExpireAt { get; set; }
+    public DateTime? ExpireAt { get; set; }
```
[the remark this answers is not in the document]
Comment (Thomas-Avery) [reply src/Api/SecretManagerFeatures/Models/Request/AccessTokenCreateRequestModel.cs:23]: Resolved in https://github.com/bitwarden/server/pull/2450/commits/ed555b756efce755aabbe9819b75194204b925fc

File: util/Migrator/DbScripts/2022-11-29_00_ApiKey_Never_Expire.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+﻿ALTER TABLE [dbo].[ApiKey] ALTER COLUMN [ExpireAt] DATETIME2(7) NULL;
```
Comment (Hinton) [inline util/Migrator/DbScripts/2022-11-29_00_ApiKey_Never_Expire.sql:1]:
```suggestion
﻿ALTER TABLE [dbo].[ApiKey] ALTER COLUMN [ExpireAt] DATETIME2(7) NULL;
﻿
```

**Review by Hinton — APPROVED** [review]
